### PR TITLE
(fix) Track file export: various fixes

### PR DESCRIPTION
**Pull request by ronso0**
1. The overwrite dialog had confusing/duplicate buttons,and clicking Skip would cancel the dialog
   before:
   export-dlg-before

   now: 
   export-dlg-now

2. Crate: use the right-clicked carte (no the selected crate, which might be null)
3. abort track export if no tracks are selected (empty crate/playlist), previously the progress dialog would be empty and remain open forever

### CODE REVIEW

**Comment by daschuer** [comment]
In the corresponding Nemo File Manager dialog we have checkbox for all files and only three buttons. 
But that would be not a bugfix PR.

**Review by daschuer — CHANGES_REQUESTED** [review]
After fixing the build it works here. I will file a feature request for the checkbox.

File: src/library/trackset/crate/cratefeature.cpp [hunk 1]
```diff
@@ -833,10 +833,15 @@ void CrateFeature::slotExportPlaylist() {
 }
 
 void CrateFeature::slotExportTrackFiles() {
+    m_lastRightClickedIndex = index;
```
[the remark this answers is not in the document]
Comment (ronso0) [reply src/library/trackset/crate/cratefeature.cpp:836]: umm :shrug:
I made the fix in 2.5, build, then rebased onto 2.4 but didn't build again, so my guess is this is a wronlgy resolved conflict.
Interesting that CI can build it.

File: src/library/export/trackexportdlg.cpp [hunk 2]
```diff
@@ -68,29 +68,30 @@ void TrackExportDlg::slotAskOverwriteMode(
...
+    QPushButton* pSkipAll = question_box.addButton(
+            tr("Skip &All"), QMessageBox::NoRole);
+    QPushButton* pOverwrite = question_box.addButton(
+            tr("&Overwrite"), QMessageBox::YesRole);
+    QPushButton* pOverwriteAll = question_box.addButton(
+            tr("Over&write All"), QMessageBox::YesRole);
+    question_box.setDefaultButton(pSkip);
+
+    question_box.exec();
+    auto pBtn = question_box.clickedButton();
```
Comment (daschuer) [inline src/library/export/trackexportdlg.cpp:84]: clang-tidy is suggesting: 
```suggestion
    auto* pBtn = question_box.clickedButton();
```

Comment (ronso0) [reply src/library/export/trackexportdlg.cpp]: :+1:

File: src/library/export/trackexportworker.cpp [hunk 3]
```diff
@@ -26,10 +26,13 @@ QMap<QString, mixxx::FileInfo> createCopylist(const TrackPointerList& tracks) {
...
     QMap<QString, mixxx::FileInfo> copylist;
     for (const auto& pTrack : tracks) {
+        VERIFY_OR_DEBUG_ASSERT(pTrack != nullptr) {
+            continue;
+        }
         auto fileInfo = pTrack->getFileInfo();
         if (fileInfo.resolveCanonicalLocation().isEmpty()) {
             qWarning()
-                    << "File not found or inaccessible while exporting"
+                    << "TrackExportWorker: File not found or inaccessible while exporting"
```
Comment (Swiftb0y) [inline src/library/export/trackexportworker.cpp:35]: wbout a QLoggingCategory instead of the prefix?

Comment (ronso0) [reply src/library/export/trackexportworker.cpp:35]: sure. I decided on mixxx::Logger though

**Review by Swiftb0y — APPROVED** [review]
LGTM. wdyt @daschuer?

**Review by daschuer — APPROVED** [review]
Works and LGTM. Thank you.

**Pull request merged**